proof_gen/src/lib.rs

```diff
@@ -134,6 +134,17 @@
 //!     assert!(verifier_state.verify(block_proof.intern).is_ok());
 //! ```
 
+#[cfg_attr(
+    not(any(feature = "polygon_pos", feature = "cdk_erigon")),
+    cfg(feature = "eth_mainnet")
+)]
```

Comment:
I don't really understand what is the purpose of this conditional attribute? If not `polygon_pos` neither `cdk_erigon` are set, compile the above `compile_error` only if `eth_mainet` is set? I think you could remove it, leave only second `cfg`

Reply:
No, this `cfg_attr` will _set_ the feature to `eth_mainnet` if no custom chain feature flag is set. This is to ensure we do have at least one targeted network.